### ISSUE TITLE
Inconsistency in number of scanned files

### ISSUE THREAD

**Issue by james-stocks** [rubocop-hq/rubocop#4654]
Offshot from https://github.com/bbatsov/rubocop/issues/4640

## Expected behavior

When running `rubocop -d` you see details of which files are being scanned, so you understand how rubocop determined `2 files inspected, no offenses detected`

## Actual behavior

In some cases, rubocop appears to scan more files than it eventually reports have been inspected

## Steps to reproduce the problem

Using Ruby 2.3 and Rubocop 0.49.1
 * Clone https://github.com/james-stocks/rubocop_bug_demo
 * (in root directory) bundle install --path vendor/bundle
 * (in testing directory) bundle install --path vendor/bundle
 * (in root directory) bundle exec rubocop -d
 
 The output indicates 3 files are scanned (lines starting "Scanning") but the summary is "2 files inspected, no offenses detected"
  
 ```
 [centos@test-development rubocop_test]$ bundle exec rubocop -d
For /home/centos/rubocop_test: configuration from /home/centos/rubocop_test/.rubocop.yml
Default configuration from /home/centos/rubocop_test/vendor/bundle/ruby/2.3.0/gems/rubocop-0.49.1/config/default.yml
Inheriting configuration from /home/centos/rubocop_test/vendor/bundle/ruby/2.3.0/gems/rubocop-0.49.1/config/enabled.yml
Inheriting configuration from /home/centos/rubocop_test/vendor/bundle/ruby/2.3.0/gems/rubocop-0.49.1/config/disabled.yml
Inspecting 4 files
Scanning /home/centos/rubocop_test/Gemfile
.Scanning /home/centos/rubocop_test/testing/Gemfile
.Scanning /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.irbrc
For /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0: configuration from /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop.yml
Inheriting configuration from /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop_todo.yml


2 files inspected, no offenses detected
Error: The `Style/TrailingComma` cop no longer exists. Please use `Style/TrailingCommaInLiteral` and/or `Style/TrailingCommaInArguments` instead.
(obsolete configuration found in /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop_todo.yml, please update it)
The `Style/SingleSpaceBeforeFirstArg` cop has been renamed to `Layout/SpaceBeforeFirstArg`.
(obsolete configuration found in /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop_todo.yml, please update it)
The `Layout/SpaceAfterControlKeyword` cop has been removed. Please use `Layout/SpaceAroundKeyword` instead.
(obsolete configuration found in /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop_todo.yml, please update it)
The `Style/SpaceAfterControlKeyword` cop has been removed. Please use `Layout/SpaceAroundKeyword` instead.
(obsolete configuration found in /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop_todo.yml, please update it)
The `Style/MethodCallParentheses` cop has been renamed to `Style/MethodCallWithoutArgsParentheses`.
(obsolete configuration found in /home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.rubocop_todo.yml, please update it)
Finished in 0.5497931893914938 seconds
 ```
(**NOTE:**  I believe the obsolete configuration warnings are aside from the issue; although in practice I would exclude testing/**/.* to exclude the file causing these warnings, and in that case the inconsistency stops. It appears that the affected file is `/home/centos/rubocop_test/testing/vendor/bundle/ruby/2.3.0/gems/fog-digitalocean-0.3.0/.irbrc`  because that file does not have a . character following it in the output?)

## RuboCop version

```
[centos@test-development rubocop_test]$ bundle exec rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.3.0 x86_64-linux)
```

**Comment by james-stocks** [comment]
It looks like it's the fog-digitalocean-0.3.0/.irbrc file (which is this file: https://github.com/fog/fog-digitalocean/blob/master/.irbrc )
If I rubocop only that file; rubocop tells me it's scanning 1 file but ends with *0 files inspected*

I think when rubocop reaches that file, it picks up on the rubocop config in that folder (this file https://github.com/fog/fog-digitalocean/blob/master/.rubocop.yml ) - at this point it stops scanning files and only reports on previously scanned files. If I remove this config file then rubocop finishes as expected.

Of course, I don't really want to scan these files as they're inside my bundle folder; and if I exclude the bundle folder then there's no problem - is this still a valid/interesting bug?

**Comment by jonas054** [comment]
I can't reproduce it. I get:
```
 ~/dev$ git clone https://github.com/james-stocks/rubocop_bug_demo
Klonar till "rubocop_bug_demo"...
remote: Enumerating objects: 14, done.
remote: Total 14 (delta 0), reused 0 (delta 0), pack-reused 14
Packar upp objekt: 100% (14/14), klart.
 ~/dev$ cd rubocop_bug_demo/
master ~/dev/rubocop_bug_demo$ bundle install --path vendor/bundle
Fetching gem metadata from https://rubygems.org/...............
Resolving dependencies...
Fetching rake 12.3.1
Installing rake 12.3.1
Fetching ast 2.4.0
Installing ast 2.4.0
Using bundler 1.16.2
Fetching parallel 1.12.1
Installing parallel 1.12.1
Fetching parser 2.5.3.0
Installing parser 2.5.3.0
Fetching powerpack 0.1.2
Installing powerpack 0.1.2
Fetching rainbow 2.2.2
Installing rainbow 2.2.2 with native extensions
Fetching ruby-progressbar 1.10.0
Installing ruby-progressbar 1.10.0
Fetching unicode-display_width 1.4.0
Installing unicode-display_width 1.4.0
Fetching rubocop 0.49.1
Installing rubocop 0.49.1
Bundle complete! 1 Gemfile dependency, 10 gems now installed.
Bundled gems are installed into `./vendor/bundle`
master ~/dev/rubocop_bug_demo$ cd testing/
master ~/dev/rubocop_bug_demo/testing$ bundle install --path vendor/bundle
Fetching gem metadata from https://rubygems.org/.............
Resolving dependencies...
Fetching CFPropertyList 2.3.6
Installing CFPropertyList 2.3.6
Fetching concurrent-ruby 1.0.5
Installing concurrent-ruby 1.0.5
Fetching i18n 0.9.5
Installing i18n 0.9.5
Fetching minitest 5.11.3
Installing minitest 5.11.3
Fetching thread_safe 0.3.6
Installing thread_safe 0.3.6
Fetching tzinfo 1.2.5
Installing tzinfo 1.2.5
Fetching activesupport 4.2.10
Installing activesupport 4.2.10
Fetching public_suffix 3.0.3
Installing public_suffix 3.0.3
Fetching addressable 2.5.2
Installing addressable 2.5.2
Fetching ansi 1.5.0
Installing ansi 1.5.0
Fetching ast 2.4.0
Installing ast 2.4.0
Fetching extlib 0.9.16
Installing extlib 0.9.16
Fetching multi_json 1.13.1
Installing multi_json 1.13.1
Fetching autoparse 0.3.3
Installing autoparse 0.3.3
Fetching aws-eventstream 1.0.1
Installing aws-eventstream 1.0.1
Fetching aws-partitions 1.107.0
Installing aws-partitions 1.107.0
Fetching aws-sigv4 1.0.3
Installing aws-sigv4 1.0.3
Fetching jmespath 1.4.0
Installing jmespath 1.4.0
Fetching aws-sdk-core 3.36.0
Installing aws-sdk-core 3.36.0
Fetching aws-sdk-ec2 1.54.0
Installing aws-sdk-ec2 1.54.0
Fetching beaker-abs 0.5.0
Installing beaker-abs 0.5.0
Fetching stringify-hash 0.0.2
Installing stringify-hash 0.0.2
Fetching beaker-aws 0.7.0
Installing beaker-aws 0.7.0
Fetching excon 0.62.0
Installing excon 0.62.0
Fetching docker-api 1.34.2
Installing docker-api 1.34.2
Fetching beaker-docker 0.4.0
Installing beaker-docker 0.4.0
Fetching multipart-post 2.0.0
Installing multipart-post 2.0.0
Fetching faraday 0.15.3
Installing faraday 0.15.3
Fetching jwt 2.1.0
Installing jwt 2.1.0
Fetching memoist 0.16.0
Installing memoist 0.16.0
Fetching os 1.0.0
Installing os 1.0.0
Fetching signet 0.11.0
Installing signet 0.11.0
Fetching googleauth 0.7.1
Installing googleauth 0.7.1
Fetching launchy 2.4.3
Installing launchy 2.4.3
Fetching retriable 1.4.1
Installing retriable 1.4.1
Fetching google-api-client 0.8.7
Installing google-api-client 0.8.7
Fetching beaker-google 0.2.0
Installing beaker-google 0.2.0
Fetching beaker-hiera 0.1.1
Installing beaker-hiera 0.1.1
Fetching deep_merge 1.2.1
Installing deep_merge 1.2.1
Fetching beaker-hostgenerator 1.1.19
Installing beaker-hostgenerator 1.1.19
Fetching builder 3.2.3
Installing builder 3.2.3
Fetching formatador 0.2.5
Installing formatador 0.2.5
Fetching mime-types-data 3.2018.0812
Installing mime-types-data 3.2018.0812
Fetching mime-types 3.2.2
Installing mime-types 3.2.2
Fetching fog-core 2.1.2
Installing fog-core 2.1.2
Fetching fog-json 1.2.0
Installing fog-json 1.2.0
Fetching ipaddress 0.8.3
Installing ipaddress 0.8.3
Fetching fog-openstack 1.0.5
Installing fog-openstack 1.0.5
Fetching beaker-openstack 0.2.0
Installing beaker-openstack 0.2.0
Fetching in-parallel 0.1.17
Installing in-parallel 0.1.17
Fetching ruby-ll 2.1.2
Installing ruby-ll 2.1.2 with native extensions
Fetching oga 2.15
Installing oga 2.15 with native extensions
Fetching beaker-puppet 0.17.1
Installing beaker-puppet 0.17.1
Fetching beaker-vagrant 0.5.0
Installing beaker-vagrant 0.5.0
Fetching beaker-vmpooler 1.3.0
Installing beaker-vmpooler 1.3.0
Fetching fission 0.5.0
Installing fission 0.5.0
Using json 2.1.0
Fetching mini_portile2 2.3.0
Installing mini_portile2 2.3.0
Fetching nokogiri 1.8.5
Installing nokogiri 1.8.5 with native extensions
Fetching trollop 2.9.9
Installing trollop 2.9.9
Fetching rbvmomi 1.13.0
Installing rbvmomi 1.13.0
Fetching beaker-vmware 0.3.0
Installing beaker-vmware 0.3.0
Fetching beaker-vcloud 0.4.0
Installing beaker-vcloud 0.4.0
Fetching hocon 1.2.5
Installing hocon 1.2.5
Fetching inifile 3.0.0
Installing inifile 3.0.0
Fetching minitar 0.7
Installing minitar 0.7
Fetching net-ssh 4.2.0
Installing net-ssh 4.2.0
Fetching net-scp 1.2.1
Installing net-scp 1.2.1
Fetching open_uri_redirections 0.2.1
Installing open_uri_redirections 0.2.1
Fetching byebug 9.0.6
Installing byebug 9.0.6 with native extensions
Fetching coderay 1.1.2
Installing coderay 1.1.2
Fetching method_source 0.9.1
Installing method_source 0.9.1
Fetching pry 0.11.3
Installing pry 0.11.3
Fetching pry-byebug 3.4.3
Installing pry-byebug 3.4.3
Fetching rb-readline 0.5.5
Installing rb-readline 0.5.5
Fetching rsync 1.0.9
Installing rsync 1.0.9
Fetching thor 0.20.0
Installing thor 0.20.0
Fetching beaker 3.37.0
Installing beaker 3.37.0
Using bundler 1.16.2
Bundle complete! 1 Gemfile dependency, 79 gems now installed.
Bundled gems are installed into `./vendor/bundle`
Post-install message from trollop:
!    The 'trollop' gem has been deprecated and has been replaced by 'optimist'.
!    See: https://rubygems.org/gems/optimist
!    And: https://github.com/ManageIQ/optimist
Post-install message from minitar:
The `minitar` executable is no longer bundled with `minitar`. If you are
expecting this executable, make sure you also install `minitar-cli`.
master ~/dev/rubocop_bug_demo/testing$ cd ..
master ~/dev/rubocop_bug_demo$ bundle exec rubocop -d
For /home/jonas/dev/rubocop_bug_demo: configuration from /home/jonas/dev/rubocop_bug_demo/.rubocop.yml
Default configuration from /home/jonas/dev/rubocop_bug_demo/vendor/bundle/ruby/2.5.0/gems/rubocop-0.49.1/config/default.yml
Inheriting configuration from /home/jonas/dev/rubocop_bug_demo/vendor/bundle/ruby/2.5.0/gems/rubocop-0.49.1/config/enabled.yml
Inheriting configuration from /home/jonas/dev/rubocop_bug_demo/vendor/bundle/ruby/2.5.0/gems/rubocop-0.49.1/config/disabled.yml
Inspecting 2 files
Scanning /home/jonas/dev/rubocop_bug_demo/Gemfile
.Scanning /home/jonas/dev/rubocop_bug_demo/testing/Gemfile
.

2 files inspected, no offenses detected
Finished in 0.22628895900561474 seconds
```
And looking at the source code for RuboCop 0.49.1 it makes perfect sense that nothing under vendor is inspected. Since we have
```yaml
AllCops:
  TargetRubyVersion: 2.1.9
  Exclude:
    # bundled gems
    - vendor/**/*
    # testing gems
    - testing/vendor/**/*
```
in `.rubocop.yml` in the root directory, the vendor directory is excluded and not searched for files. So I can't explain how @james-stocks got a different result.

I'm closing the issue and if anyone can reproduce the problem, please re-open with details.
